xlators/features/shard/src/shard.c

```diff
@@ -3642,6 +3644,45 @@ shard_lookup_marker_entry(xlator_t *this, shard_local_t *local,
     return ret;
 }
 
+static int
+shard_nameless_lookup_base_file(xlator_t *this, char *gfid)
+{
+    int ret = 0;
+    loc_t loc = {
+        0,
+    };
+    dict_t *xattr_req = dict_new();
+    if (!xattr_req) {
+        ret = -ENOMEM;
+        goto out;
+    }
+
+    loc.inode = inode_new(this->itable);
+    if (loc.inode == NULL) {
+        ret = -ENOMEM;
+        goto out;
+    }
+
+    ret = gf_uuid_parse(gfid, loc.gfid);
+    if (ret < 0)
+        goto out;
+
+    ret = dict_set_uint32(xattr_req, GF_UNLINKED_LOOKUP, 1);
+    if (ret < 0)
+        goto out;
+
+    ret = syncop_lookup(FIRST_CHILD(this), &loc, NULL, NULL, xattr_req, NULL);
+    if (ret < 0)
+        goto out;
+
+out:
+    if (xattr_req)
+        dict_unref(xattr_req);
+    loc_wipe(&loc);
+
+    return ret;
```

Comment:
`ret` has different meanings. It can contain an error code or be simply -1 in case of error. I prefer to use error codes, but given that we depend on functions that only return -1 and we can't change them now, I think we should always return -1 in case of error.

Reply:
okay. Makes sense. I will return -1 instead of specific error codes.